components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -742,6 +742,29 @@ register_variables(const std::string& filename,
 
     register_variable(filename, name, name, units, vec_of_dims,
                       "real",fp_precision, io_decomp_tag);
+
+    // Add any extra attributes for this variable, examples include:
+    //   1. A list of subfields associated with a field group output
+    //   2. A CF longname (TODO)
+    // First check if this is a field group w/ subfields.
+    const auto& children = field.get_header().get_children();
+    if (children.size()>0) {
+      // This field is a parent to a set of subfields
+      std::stringstream children_list;
+      children_list << "[ ";
+      int it = 0;
+      for (const auto& child_weak : children) {
+        auto child = child_weak.lock();
+        it ++;
+        children_list << child->get_identifier().name().c_str();
+        if (it < children.size()) {
+          children_list << ", ";
+        }
+      }
+      children_list << " ]";
```

Comment:
Instead of using the it variable (which makes the impl more puzzling to read), consider a simpler approach
```
std::string children_list = "[";
for (auto ch_w : children) {
  auto child = ch_w.lock();
  children_list += child->get_identifier().name() + ", ";
}
// Remove last ", "
children_list.pop_back();
children_list.pop_back();
set_variable_attribute_c2f(...);
```

Reply:
Excellent, I had just the same thought last night before going to bed.